File: go-admin-ui/src/api/scblines.js
import request from '@/utils/request'

// 查询ScbLines列表
export function listScbLines(query) {
  return request({
    url: '/api/v1/scblinesList',
    method: 'get',
    params: query
  })
}

// 查询ScbLines详细
export function getScbLines(id) {
  return request({
    url: '/api/v1/scblines/' + id,
    method: 'get'
  })
}

// 新增ScbLines
export function addScbLines(data) {
  return request({
    url: '/api/v1/scblines',
    method: 'post',
    data: data
  })
}

// 修改ScbLines
export function updateScbLines(data) {
  return request({
    url: '/api/v1/scblines',
    method: 'put',
    data: data
  })
}

// 删除ScbLines
export function delScbLines(id) {
  return request({
    url: '/api/v1/scblines/' + id,
    method: 'delete'
  })
}

export function carsTreeselect(lineId) {
  return request({
    url: '/api/v1/carsTreeselect/' + lineId,
    method: 'get'
  })
}
// 获取所有线路
export function getAllLines() {
  return request({
    url: '/api/v1/getAllLines',
    method: 'get'
  })
}
// 获取线路的所有车辆
export function getLineCars(lineId) {
  return request({
    url: '/api/v1/getLines/cars?id=' + lineId,
    method: 'get'
  })
}
// 获取线路的所有站点
export function getLineSites(lineId) {
  return request({
    url: '/api/v1/getLines/sites?id=' + lineId,
    method: 'get'
  })
}

File: go-admin/router/scblines.go
package router

import (
	"github.com/gin-gonic/gin"
	"go-admin/apis/scblines"
	"go-admin/middleware"
	jwt "go-admin/pkg/jwtauth"
)

// 需认证的路由代码
func registerScbLinesRouter(v1 *gin.RouterGroup, authMiddleware *jwt.GinJWTMiddleware) {

	r := v1.Group("/scblines").Use(authMiddleware.MiddlewareFunc()).Use(middleware.AuthCheckRole())
	{
		r.GET("/:id", scblines.GetScbLines)
		r.POST("", scblines.InsertScbLines)
		r.PUT("", scblines.UpdateScbLines)
		r.DELETE("/:id", scblines.DeleteScbLines)

		//r.GET("/list", scblines.GetScbLinesList)
		//r.GET("/treeselect/:id", scblines.GetScbLinesTreeCarsselect)
		//r.GET("/all-lines", scblines.GetAllLines)
	}

	l := v1.Group("").Use(authMiddleware.MiddlewareFunc()).Use(middleware.AuthCheckRole())
	{
		l.GET("/scblinesList", scblines.GetScbLinesList)
		l.GET("/carsTreeselect/:id", scblines.GetScbLinesTreeCarsselect)
		//获取所有线路
		l.GET("/getAllLines", scblines.GetAllLines)
		//获取线路的所有车辆
		l.GET("/getLines/cars", scblines.GetCars)
		//获取线路的所有站点
		l.GET("/getLines/sites", scblines.GetSites)
	}

}
